crates/client/eth/src/l1_messaging.rs

```diff
@@ -62,7 +62,9 @@ pub async fn sync(
         .to_block(BlockNumberOrTag::Finalized)
         .watch()
         .await
-        .context("Failed to watch event filter")?
+        .context(
+            "Failed to watch event filter - Ensure you are using an L1 RPC endpoint that points to an archive node",
+        )?
```

Comment:
Interesting. I'm not familiar with Eth l1 non-archive nodes, what are the differences with typical archive nodes?

Reply:
L1 Full nodes only contain pruned state.